lib/vagrant/shared_helpers.rb

```diff
@@ -38,11 +38,11 @@ def self.installer_embedded_dir
     ENV["VAGRANT_INSTALLER_EMBEDDED_DIR"]
   end
 
-  # This returns whether or not 3rd party plugins should be loaded.
+  # This returns whether or not 3rd party plugins should and can be loaded.
   #
   # @return [Boolean]
   def self.plugins_enabled?
-    !ENV["VAGRANT_NO_PLUGINS"]
+    !ENV["VAGRANT_NO_PLUGINS"] && $vagrant_bundler_runtime
```

Comment:
I think this change makes sense, but of course we could also let it unmodified, and adapt the unit tests context to avoid https://travis-ci.org/mitchellh/vagrant/builds/70328862#L196.